src/odb.c

```diff
@@ -297,14 +297,9 @@ int git_odb__hashlink(git_oid *out, const char *path)
 		GIT_ERROR_CHECK_ALLOC(link_data);
 
 		read_len = p_readlink(path, link_data, size);
-		link_data[size] = '\0';
```

Comment:
is there a way to preserve this in some fashion? maybe `link_data[read_len] = '\0';` after the success check.

Reply:
We can, but it seems redundant &mdash; the buffer is being passed as `const void *` + `size_t` to git_odb_hash, which hashes that many bytes. It wouldn't be looking for the null terminator.

Is there a debugging rationale for having the null terminator anyway? Like maybe so that printing `link_data` in a debugger prints the right amount of data?

Reply:
Added `link_data[read_len] = '\0';`.

Reply:
> Is there a debugging rationale for having the null terminator anyway? Like maybe so that printing link_data in a debugger prints the right amount of data?

I'm lacking the historical context.  It's possible that we originally called a different function here that expected nul terminated string, or `git_odb_hash` has changed, etc.  I agree that this is unnecessary, but I also agree that having nul terminated strings in a debugger is helpful.  Given all of that, this seems like a nice fix.  If you wanted to attack the nul termination problem separately, I wouldn't argue, but I also don't think that it's a big problem either.  🤷‍♂️